lib/src/kbasesearchengine/search/ObjectData.java

```diff
@@ -526,5 +548,18 @@ public Builder withKeyProperty(final String key, final String property) {
             keyProps.put(key, property);
             return this;
         }
+
+        /** Adds the highlight fields to the object.
+         * @param highlight the map of fields returned from elasticsearch.
+         * @return this builder.
+         */
+        public Builder withHighlight(final String field, final ArrayList highlight) {
+            Utils.notNullOrEmpty(field, "field cannot be null or whitespace");
+            Utils.noNulls(highlight, "highlight cannot be null");
```

Comment:
Check for any whitespace only highlight entries and throw an exception if so

Reply:
is there a need to iterate through the highlights and check? b/c if elastic search highlight is returning just whitespaces its broken

Reply:
Yes, there's no guarantee that this class will always be paired with elasticsearch. A class should not consider implementation details of its dependencies in its design, because those implementation details may change.

Reply:
Also elasticsearch isn't really a dependency anyway - this class doesn't know elasticsearch exists.

Reply:
The other problem I just realized about this implementation is that because the list is passed in and returned as is, it allows the internal state to be modified from outside the class after creation, breaking the classes' immutability

Reply:
does this mean that I should return an unmodifiableMap in fn getHighlight?

Reply:
I would just build the entire structure so it's immutable in the constructor. Then you can return whatever and be safe.

The other approach I've used is instead of returning the map, have getKeys() and getValue(key) methods (appropriately named) so the value class doesn't have to expose the map.

Reply:
okay. its an unmodifiableMap in the constructor. 

As far as I understand, the current MVP wants me to chuck the whole map out so the second approach won't make sense.

Reply:
You could still do it that way, you'd just rebuild the map from the class.

Reply:
I could but isnt that what the unmodifiable map already doing in addition to not allowing you to change the outputted map? If my only use case is returning everything, then it seemed kind of unnecessary to do the rebuild myself.

Reply:
I said could, not should :P